Document how we want to treat @throws tags

fnwbr opened the pull request:
@iwiznia

iwiznia commented on php.md:
````diff
@@ -978,7 +936,31 @@ public function __construct(Report $report)
 ```
 
-- When function is deprecated in favor of an other one, the `@deprecated` *must* be used
+- PHPDocs *must* document all exceptions being thrown only in the first level using `@throws` tag
````
I don't think `first level` is clear... I would say: `all exceptions thrown directly by this method`

iwiznia commented on php.md:
````diff
@@ -1017,6 +992,15 @@ $randomVariable = PolicyStore::getID();
     }
 ```
 
+- PHPDocs *should* be used on variables, if they add value
````
I would say `can` because almost no one does this and is just a nice addition for people who use IDEs.

flodnv replied:
Now that PHP has become way smarter with types, I think the only code that needs this now is code that smells (none of our tooling can detect its type because no one knows)

flodnv commented on php.md:
````diff
@@ -1017,6 +992,15 @@ $randomVariable = PolicyStore::getID();
+
+```PHP
+$emails = array_values(array_unique(array_map(function ($reportAction) {
+    /* @var Report_Action_Item $reportAction */
````
Actually this example is typical of where you shouldn't use it. The line above needs to be
`function (Report_Action_Item  $reportAction)`

iwiznia replied:
lol, true that

fnwbr commented:
Updated; after talking to @flodnv I removed the "PHPDocs for variables" part completely - we should just not mention it since there's no case these days where you'd want to embrace someone to write code where we need those
